sysbuild: Minor MCUboot configuration fixes

nordicjm opened the pull request:
Fixes some issues when configuring MCUboot using sysbuild

nordicjm committed:
sysbuild: bootloader: Fix setting application Kconfig in MCUboot

Fixes wrongly setting configuration to generate an unsigned image
in the MCUboot configuration when this does not apply to MCUboot

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
sysbuild: mcuboot: Disable signing key in no signing mode

Prevents allowing the user to enter a signing key when the signing
mode is set to hash check only without signatures

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

MaureenHelm commented:
ping @tejlmand @57300

57300 commented on share/sysbuild/image_configurations/BOOTLOADER_image_default.cmake:
```diff
@@ -12,6 +12,7 @@ set(keytypes CONFIG_BOOT_SIGNATURE_TYPE_NONE
 
 if(SB_CONFIG_BOOT_SIGNATURE_TYPE_NONE)
   set(keytype CONFIG_BOOT_SIGNATURE_TYPE_NONE)
+  set_config_bool(${ZCMAKE_APPLICATION} CONFIG_MBEDTLS n)
```
> Fixes an issue whereby a project is configured then the configuration is changed multiple times to work with signatures to without whereby a conflict between tinycrypt and MBEDTLS would appear

How can we reproduce this?

nordicjm replied:
Configure smp_svr using sysbuild, run `ninja sysbuild_menuconfig` and change signature type to none, then build

57300 replied:
Thanks, I'm not sure why I couldn't reproduce it at all before. But now I can see that this is not a sysbuild-specific issue, because I can: 

1. Run `cmake -S$(west topdir)/bootloader/mcuboot/boot/zephyr -GNinja -DBOARD=nrf52840dk_nrf52840`
2. Set `CONFIG_BOOT_SIGNATURE_TYPE_NONE=y`
3. Build

and get the same error. So, shouldn't this be fixed in MCUboot proper?

nordicjm replied:
It can't, mbedlts will default to on with the initial config, when `menuconfig` is ran and the setting is changed, it is no longer using the `default` value but the saved value from the .config file, so it is not able to know that it can be un-selected.

57300 replied:
How about selecting `MBEDTLS_PROMPTLESS` in MCUboot?

nordicjm replied:
Have updated

nordicjm committed:
west.yml: MCUboot synchronization from upstream

Update Zephyr fork of MCUboot to revision:
  47b34362552835621bc289d53d2127691088cb7c

Brings following Zephyr relevant fixes:
 - 47b34362 zephyr: kconfig: Prevent MBEDTLS selection when
   tinycrypt is used
 - cd82f8bf boot: zephyr: add support for lpcxpresso55s28

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm commented:
@erwango CI looks to be broken on stm32 lpuart?

erwango commented:
> @erwango CI looks to be broken on stm32 lpuart?

Yes, sorry for that. Fixed by https://github.com/zephyrproject-rtos/zephyr/pull/65532
